Now I remember, the main why this is necessary, is that this
> Ensure all linting tests pass, see here for how to run them

links to this page, without any further link to how to set up pre-commits.
here you go:
![docs](https://user-images.githubusercontent.com/33122845/132351333-58ac6db3-c247-4b2b-b0a8-3d8f583a42d3.jpg)
The period is fixed also.
The build process takes very long on my machine (multiple minutes). Is there a way to make it go faster?
Also, I got the following errors while building, unrelated to what I did:
<details>
```
WARNING: 
>>>-------------------------------------------------------------------------
Exception in /mnt/data/Git/pd_backup/doc/source/whatsnew/v0.9.1.rst at block ending on line 81
Specify :okexcept: as an option in the ipython:: block to suppress this message
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
<ipython-input-6-b548a30a7bce> in <module>
----> 1 df = DataFrame(np.random.randn(5, 3), columns = ['A','B','C'])

NameError: name 'DataFrame' is not defined

<<<-------------------------------------------------------------------------
/mnt/data/Git/pd_backup/doc/source/whatsnew/v1.3.0.rst:1250: WARNING: Cannot find contributors for range 'v1.2.5..v1.3.0': Cmd('git') failed due to: exit code(128)
  cmdline: git log --grep=Co-authored --pretty=%b v1.2.5..v1.3.0
  stderr: 'fatal: ambiguous argument 'v1.2.5..v1.3.0': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]''
/mnt/data/Git/pd_backup/doc/source/whatsnew/v1.3.1.rst:65: WARNING: Cannot find contributors for range 'v1.3.0..v1.3.1': Cmd('git') failed due to: exit code(128)
  cmdline: git log --grep=Co-authored --pretty=%b v1.3.0..v1.3.1
  stderr: 'fatal: ambiguous argument 'v1.3.0..v1.3.1': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]''
/mnt/data/Git/pd_backup/doc/source/whatsnew/v1.3.2.rst:65: WARNING: Cannot find contributors for range 'v1.3.1..v1.3.2': Cmd('git') failed due to: exit code(128)
  cmdline: git log --grep=Co-authored --pretty=%b v1.3.1..v1.3.2
  stderr: 'fatal: ambiguous argument 'v1.3.1..v1.3.2': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]''
/mnt/data/Git/pd_backup/doc/source/whatsnew/v1.3.3.rst:78: WARNING: Cannot find contributors for range 'v1.3.2..v1.3.3|HEAD': Cmd('git') failed due to: exit code(128)
  cmdline: git log --grep=Co-authored --pretty=%b v1.3.2..HEAD
  stderr: 'fatal: ambiguous argument 'v1.3.2..HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]''
WARNING: 
>>>-------------------------------------------------------------------------
Exception in /mnt/data/Git/pd_backup/doc/source/whatsnew/v0.18.0.rst at block ending on line 1006
Specify :okexcept: as an option in the ipython:: block to suppress this message
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-108-c90c8ac4f6a1> in <module>
----> 1 s.between_time("7:00am", "9:00am")

/mnt/data/Git/pd_backup/pandas/core/generic.py in between_time(self, start_time, end_time, include_start, include_end, axis)
   7681             raise TypeError("Index must be DatetimeIndex")
   7682 
-> 7683         indexer = index.indexer_between_time(
   7684             start_time, end_time, 
```
</details>

--- a/.github/PULL_REQUEST_TEMPLATE.md
+++ b/.github/PULL_REQUEST_TEMPLATE.md
@@ -1,4 +1,4 @@
 - [ ] closes #xxxx
 - [ ] tests added / passed
-- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
+- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
 - [ ] whatsnew entry

--- a/doc/source/development/contributing.rst
+++ b/doc/source/development/contributing.rst
@@ -334,7 +334,7 @@ can comment::
 on that pull request. This will trigger a workflow which will autofix formatting
 errors. To automatically fix formatting errors on each commit you make, you can
 set up pre-commit yourself. First, create a Python :ref:`environment
-<contributing.python-env>` , and then set up :ref:`pre-commit <contributing.pre-commit>`
+<contributing.python-env>` , and then set up :ref:`pre-commit <contributing.pre-commit>`.
 
 Delete your merged branch (optional)
 ------------------------------------

--- a/doc/source/development/contributing.rst
+++ b/doc/source/development/contributing.rst
@@ -332,9 +332,11 @@ can comment::
     @github-actions pre-commit
 
 on that pull request. This will trigger a workflow which will autofix formatting
-errors. To automatically fix formatting errors on each commit you make, you can
+errors.
+
+To automatically fix formatting errors on each commit you make, you can
 set up pre-commit yourself. First, create a Python :ref:`environment
-<contributing.python-env>` , and then set up :ref:`pre-commit <contributing.pre-commit>`.
+<contributing.python-env>` and then set up :ref:`pre-commit <contributing.pre-commit>`.
 
 Delete your merged branch (optional)
 ------------------------------------

--- a/doc/source/development/contributing_environment.rst
+++ b/doc/source/development/contributing_environment.rst
@@ -133,7 +133,7 @@ compiler installation instructions.
 
 Let us know if you have any difficulties by opening an issue or reaching out on `Gitter <https://gitter.im/pydata/pandas/>`_.
 
-.. _contributing.python-env
+.. _contributing.python-env:
 
 Creating a Python environment
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

